.github/actions/artifact_upload/action.yml

```diff
@@ -0,0 +1,50 @@
+name: Upload artifact
+description: Upload an encrypted zip archive as a github artifact.
+inputs:
+  path:
+    description: 'The path(s) that should be uploaded.'
+    required: true
+  name:
+    description: 'The name of the artifact.'
+    required: true
+  retention-days:
+    description: 'How the artifact should be retained.'
+    required: false
+    default: 60
+  encryption-secret:
+    description: 'The secret to use for encrypting the files.'
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Install zip
+      uses: montudor/action-zip@v1
+    - name: Find glob pattern matches
+      uses: tj-actions/glob@v17
+      id: glob
+      with:
+        files: |
+          ${{ inputs.path }}
+    - name: Create temporary directory
+      id: tempdir
+      shell: bash
+      run: echo "directory=$(mktemp -d)" >> "$GITHUB_OUTPUT"
+    - name: Create archive
+      shell: bash
+      run: |
+        # Create an archive if files exist.
+        # Don't create an archive file if no files are found
+        # and warn/error if configured that way.
+        if [ -n "${{ steps.glob.outputs.paths }}" ]
+        then
+          zip -e -P '${{ inputs.encryption-secret }}' -qq -r ${{ steps.tempdir.outputs.directory }}/archive.zip ${{ steps.glob.outputs.paths }}
```

Comment:
Won't `-e` require a password to be entered in a prompt? Also, why do we need `-r` here. We would only want to unpack one archive, right?

Reply:
`-e` specifies that the archive should be encrypted. However, it does not ask for a password since it's provided with the `-P` flag. The `-r` is there because per default, `zip` doesn't recursively include stuff in directories, so if you give it just a directory it wouldn't include any files in that directory if the `-r` flag isn't given.

Reply:
Thanks for the explanation!